File: python_programmers_school/level 1/부족한 금액 계산하기.py
def solution(price, money, count):
    total_price = price * (count*(count+1) / 2)
    if money - total_price < 0:
        return total_price - money
    return 0

# 한줄 코드
# def solution(price, money, count):
#     return max(0,price*(count+1)*count//2-money)     --> 0과 오른쪽 값 중 더 큰 것을 반환!
# 오른쪽 값이 양수면 돈이 모자르다는 의미이고 0보다 크므로 그 값이 반환된다.
# 오른쪽 값이 음수면 돈이 모자르지 않다는 의미이고 0보다 작으므로 0이 반환된다.

File: javascript_programmers_school/level 1/부족한 금액 계산하기.js
function solution(price, money, count) {
  let need = 0;
  for (let cnt = 1; cnt <= count; cnt++) {
    need += price * cnt;
  }
  return money - need >= 0 ? 0 : need - money;
}
// 가지고 있는 돈이 필요한 돈보다 많을 경우도 있을 수 있으므로 return need - money 할 게 아니라 옆 경우도 생각해 주어야 함.

// 다른 사람의 코드 (가우스 공식 이용, 등차수열의 합 생각)
function solution(price, money, count) {
  const tmp = (price * count * (count + 1)) / 2 - money;
  return tmp > 0 ? tmp : 0;
}
